hathor/transaction/base_transaction.py

```diff
@@ -481,22 +496,39 @@ def validate_basic(self, skip_block_weight_verification: bool = False) -> bool:
         meta.validation = ValidationState.BASIC
         return True
 
-    def validate_full(self, skip_block_weight_verification: bool = False) -> bool:
+    def validate_full(self, skip_block_weight_verification: bool = False, sync_checkpoints: bool = False) -> bool:
```

Comment:
What does `sync_checkpoints` do here?

Reply:
When we're syncing checkpoints we want to mark this as `CHECKPOINT_FULL` so we still remember that this was validated from a checkpoint and not "manually".

Reply:
Fixed in #292 .